modules/component/src/componentNext.ts

```diff
@@ -98,13 +86,38 @@ export class ComponentNext<P1 extends ComponentProps> {
     )
   }
 
+  /**
+   * Creates a new component with initial state as undefined
+   *
+   * ```typescript
+   * import {ComponentNext} from '@action-land/component'
+   *
+   * const component = ComponentNext.empty // creates new component with initial state as undefined
+   * ```
+   */
   static get empty(): ComponentNext<{
     iState: undefined
     oView: void
   }> {
     return ComponentNext.lift(undefined)
   }
 
+  /**
+   * Adds ability to transform component's state matching action type
+   *
+   * ```typescript
+   * import {ComponentNext} from '@action-land/component'
+   *
+   * const component = ComponentNext.lift({count: 100})
+   *  .matchR('add', (value: number, state) => ({count: state.count + value}))
+   *  // Adds behaviour to handle action of type Action<number, 'add'>
+   * ```
+   * @typeparam T Action type being handled by operator
+   * @typeparam V Value of action being handled by operator
+   * @typeparam oState2 New state returned by the cb function
+   * @param type Action type for which we want to add behaviour
+   * @param cb Function that takes in a action's value and a state and returns a new state
```

Comment:
could potentially be shorted to: `transformation function that returns a new state`